tests/phpunit/CRM/Contribute/BAO/ContributionTest.php

```diff
@@ -1195,4 +1195,45 @@ public function createContributionWithTax() {
     return array($contribution, $financialAccount);
   }
 
+  /**
+   * function to test card_type and pan truncation.
+   */
+  public function testCardTypeAndPanTruncation() {
+    $contactId = $this->individualCreate();
+    $params = array(
+      'contact_id' => $contactId,
+      'receive_date' => '2016-01-20',
+      'total_amount' => 100,
+      'financial_type_id' => 1,
+      'payment_instrument_id' => 1,
+      'card_type' => 'Visa',
+      'pan_truncation' => 4567,
+    );
+    $contribution = CRM_Contribute_BAO_Contribution::create($params);
```

Comment:
@eileenmcnaughton since its a CRM unit test i added BAO method to test if pan truncation is updated correctly. I have also added the same test using api also.